src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -159,41 +158,35 @@ public void Init(HttpApplication application)
                     if (request == null)
                         return;
 
-                    Logger.Write(_requestLoggingLevel, "HTTP {Method} {RawUrl} responded {StatusCode} in {ElapsedMilliseconds}ms", 
-                        request.HttpMethod, 
-                        request.RawUrl, 
-                        application.Response.StatusCode, 
-                        stopwatch.ElapsedMilliseconds);
+                    var error = application.Server.GetLastError();
+                    var level = error != null ? LogEventLevel.Error : _requestLoggingLevel;
 
-                    if (ShouldLogFormData())
+                    var logger = Logger;
+                    if (ShouldLogFormData)
                     {
                         var form = request.Unvalidated.Form;
-
                         if (form.HasKeys())
                         {
                             var formData = form.AllKeys.SelectMany(k => (form.GetValues(k) ?? new string[0]).Select(v => new { Name = k, Value = FilterPasswords(k, v) }));
-                            Logger.Write(_formDataLoggingLevel, "Client provided {@FormData}", formData);
+                            logger = logger.ForContext("FormData", formData, true);
                         }
                     }
-                }                
-            };
 
-            application.Error += (sender, args) =>
-            {
-                if (_isEnabled)
-                {
-                    var ex = ((HttpApplication)sender).Server.GetLastError();
-                    Logger.Error(ex, "Error caught in global handler: {ExceptionMessage}", ex.Message);
-                }
+                    logger.Write(
+                        level,
+                        error,
+                        "HTTP {Method} {RawUrl} responded {StatusCode} in {ElapsedMilliseconds}ms", 
+                        request.HttpMethod, 
+                        request.RawUrl, 
+                        application.Response.StatusCode, 
+                        stopwatch.ElapsedMilliseconds);
+                }                
             };
         }
 
-        static bool ShouldLogFormData()
-        {
-            return Logger.IsEnabled(_formDataLoggingLevel) 
-                && (LogPostedFormData == LogPostedFormDataOption.Always
-                || (LogPostedFormData == LogPostedFormDataOption.OnlyOnError && HttpContext.Current.Response.StatusCode >= 500));
-        }
+        static bool ShouldLogFormData => Logger.IsEnabled(_formDataLoggingLevel) &&
```

Comment:
Yes, I'm just digging into it myself and finding some nice little conveniences such as this one.